sdk/core/azure-core/test/ut/curl_connection_pool.cpp

```diff
@@ -30,9 +31,13 @@ namespace Azure { namespace Core { namespace Test {
 
     TEST(CurlConnectionPool, connectionPoolTest)
     {
-      Azure::Core::Http::_detail::CurlConnectionPool::ClearIndex();
+      {
+        std::lock_guard<std::mutex> lock(
+            CurlConnectionPool::g_curlConnectionPool.ConnectionPoolMutex);
```

Comment:
Why is taking a lock required here?

Reply:
In the case where many tests run in the same application, if this test is ever run after running some other libcurl tests which previously created a connection and returned to the pool, then the CleanPoolThread is already running and might crash if it is checking a connection and then we clear the connections here.

Reply:
Ah I see. Good to know. This explanation would be great as a comment in source :)